File: frontend/app/lib/types/user-settings.ts
import { z } from "zod";

export const UserSettingsSchema = z.object({
  default_vcpus: z.number().nullable(),
  default_mem: z.number().nullable(),
  default_disk_size: z.number().nullable(),
  default_os: z.string().nullable(),
  default_keyboard_layout: z.string().nullable(),
  timezone: z.string(),
});

export type UserSettings = z.infer<typeof UserSettingsSchema>;

export interface UpdateUserSettingsPayload {
  default_vcpus?: number | null;
  default_mem?: number | null;
  default_disk_size?: number | null;
  default_os?: string | null;
  default_keyboard_layout?: string | null;
  timezone?: string | null;
}

File: backend/app/models/user_settings.py
from typing import Optional
from pydantic import BaseModel


class UserSettingsResponse(BaseModel):
    default_vcpus: Optional[int] = None
    default_mem: Optional[int] = None
    default_disk_size: Optional[int] = None
    default_os: Optional[str] = None
    default_keyboard_layout: Optional[str] = None
    timezone: str = "auto"


class UpdateUserSettingsRequest(BaseModel):
    default_vcpus: Optional[int] = None
    default_mem: Optional[int] = None
    default_disk_size: Optional[int] = None
    default_os: Optional[str] = None
    default_keyboard_layout: Optional[str] = None
    timezone: Optional[str] = None
